sklearn/model_selection/_validation.py

```diff
@@ -759,17 +767,27 @@ def learning_curve(estimator, X, y, groups=None,
 
     parallel = Parallel(n_jobs=n_jobs, pre_dispatch=pre_dispatch,
                         verbose=verbose)
+    if shuffle:
+        rng = check_random_state(random_state)
+
     if exploit_incremental_learning:
         classes = np.unique(y) if is_classifier(estimator) else None
         out = parallel(delayed(_incremental_fit_estimator)(
-            clone(estimator), X, y, classes, train, test, train_sizes_abs,
-            scorer, verbose) for train, test in cv.split(X, y, groups))
+            clone(estimator), X, y, classes, train if not shuffle else
+            rng.permutation(train), test, train_sizes_abs, scorer, verbose)
```

Comment:
Can we please pull the generator out, perhaps into a separate function, to be shared by the incremental and non-incremental cases?